check_puppetdb_node.rb

```diff
@@ -10,24 +10,48 @@
 require 'time'
 require 'date'
 
-def api_call(protocol="http", host="localhost", port="8080", query)
+def api_call(query, opt = {})
+  protocol     = opt.has_key?(:ssl) ? 'https' : 'http'
+  host         = opt[:host] || 'localhost'
+  port         = opt[:port] || opt.has_key?(:ssl) ? 8081 : 8080
   uri          = URI("#{protocol}://#{host}:#{port}/#{query}")
   http         = Net::HTTP.new(uri.host, uri.port)
-  http.use_ssl = true  if protocol == 'https'
-  http.use_ssl = false if protocol == 'http'
-  request      = Net::HTTP::Get.new(uri.request_uri, {'Accept' => 'application/json'})
-  response     = http.request(request)
+  http.use_ssl = opt.has_key?(:ssl)
+  if http.use_ssl?
+    require 'openssl/x509'
+    require 'openssl/pkey'
+
+    http.ca_file = opt[:ca] if opt.has_key?(:ca)
```

Comment:
Needs tests if this adds CRL verify support:
```suggestion
    if opt.has_key?(:ca)
      cert_store = OpenSSL::X509::Store.new
      cert_store.add_file(opt[:ca])
      cert_store.add_crl(OpenSSL::X509::CRL.new(File.read(opt[:crl]))) if opt.has_key?(:crl)
      http.cert_store = cert_store
    end
```

Reply:
Tested and added.